pkg/status/status_test.go

```diff
@@ -64,65 +38,121 @@ var _ = Describe("Status pkg UTs", func() {
 		defer os.Remove(f.Name())
 		st := New(f.Name())
 
-		st.SetReady("anykey", false, "reason1")
+		// Function to get the file mod tinw
+		modTimeFn := func() time.Time {
+			prevStat, err := os.Stat(f.Name())
+			Expect(err).NotTo(HaveOccurred())
+			return prevStat.ModTime()
+		}
 
 		By("status file should return proper Status object", func() {
+			st.SetReady("anykey", false, "reason1")
+
 			readSt, err := ReadStatusFile(f.Name())
 			Expect(err).NotTo(HaveOccurred())
 
-			Expect(readSt.Readiness).Should(HaveKeyWithValue("anykey",
+			Expect(readSt.Readiness).To(HaveKeyWithValue("anykey",
 				ConditionStatus{Ready: false, Reason: "reason1"}))
-			Expect(readSt.GetReadiness()).Should(Equal(false))
+			Expect(readSt.GetReadiness()).To(Equal(false))
 		})
 
 		By("status file should be updated when reason changes", func() {
 			st.SetReady("anykey", false, "reason2")
+
+			// File should be updated, check the data.
 			readSt, err := ReadStatusFile(f.Name())
 			Expect(err).NotTo(HaveOccurred())
-
-			Expect(readSt.Readiness).Should(HaveKeyWithValue("anykey",
+			Expect(readSt.Readiness).To(HaveKeyWithValue("anykey",
 				ConditionStatus{Ready: false, Reason: "reason2"}))
-			Expect(readSt.GetReadiness()).Should(Equal(false))
+			Expect(readSt.GetReadiness()).To(Equal(false))
 		})
 
 		By("status file should be updated when set ready", func() {
 			st.SetReady("anykey", true, "")
+
+			// File should be updated, check the data.
 			readSt, err := ReadStatusFile(f.Name())
 			Expect(err).NotTo(HaveOccurred())
-
-			Expect(readSt.Readiness).Should(HaveKeyWithValue("anykey",
+			Expect(readSt.Readiness).To(HaveKeyWithValue("anykey",
 				ConditionStatus{Ready: true, Reason: ""}))
-			Expect(readSt.GetReadiness()).Should(Equal(true))
+			Expect(readSt.GetReadiness()).To(Equal(true))
 		})
 
 		By("status file should not be updated when set ready a 2nd time", func() {
-			prevStat, err := os.Stat(f.Name())
-			Expect(err).NotTo(HaveOccurred())
-
-			// Set ready again
+			prevMod := modTimeFn()
```

Comment:
I don't really think it is worth it to wrap the stat time and check in a function.  The stack that is shown if the expect in the function fails won't indicate which call it failed at.  If this function was done in a lot of places then I would be more on board with it but since it is isolated to this one `By` I don't think it is worth it.
If you feel strongly that what you have is better then could you perhaps change the function name to something like `getStatusFileModificationTime`.

Reply:
Yeah I can revert that - it was a remnant of when I had asynchronous file writing and was using Eventually and Consistently.